cli/pkg/timescaledb_secrets/create.go

```diff
@@ -0,0 +1,316 @@
+package timescaledb_secrets
+
+import (
+	"bytes"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	crand "crypto/rand"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"math/big"
+	"os"
+	"time"
+
+	"github.com/timescale/tobs/cli/pkg/k8s"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type TSDBSecretsInfo struct {
+	ReleaseName    string
+	Namespace      string
+	EnableS3Backup bool
+	TlsCert        []byte
+	TlsKey         []byte
+}
+
+func (t *TSDBSecretsInfo) CreateTimescaleDBSecrets() error {
+	// Previous helm install used to create namespace if it doesn't exist
+	// but as we are creating secrets prior to deploying tobs. We are verifying
+	// namespace if doesn't create one.
+	err := k8s.CreateNamespaceIfNotExists(t.Namespace)
+	if err != nil {
+		return err
+	}
+
+	if t.TlsCert == nil && t.TlsKey == nil {
+		err = t.createTimescaleDBCertificates()
+		if err != nil {
+			return err
+		}
+	} else {
+		fmt.Println("Using the provided TLS certificates for TimescaleDB.")
+	}
+
+	err = t.createTimescaleDBCredentials()
+	if err != nil {
+		return err
+	}
+
+	if t.EnableS3Backup {
+		err = t.createS3BackupForTimescaleDB()
+		if err != nil {
+			return err
+		}
+	} else {
+		fmt.Println("skipping to create TimescaleDB s3 backup secret as backup option is disabled.")
+	}
+
+	return nil
+}
+
+// randomPassword generates a random alphanumeric password of a given length.
+// Compatible with passwords generated by the postgres operator
+func randomPassword(n int) ([]byte, error) {
+	var passwordChars = []byte("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789")
+	b := make([]byte, n)
+	_, err := crand.Read(b)
+	if err != nil {
+		return b, fmt.Errorf("failed to generate random password %v", err)
+	}
+	for i := 0; i < n; i++ {
+		b[i] = passwordChars[b[i]/5]
```

Comment:
this should read `b[i] = passwordChars[uint8(b[i]) % len(passwordChars)]`

Reply:
Updating it to this `b[i] = passwordChars[b[i] % uint8(len(passwordChars))]` as `uint8(b[i])` is redundant conversion.